packtools/sps/validation/funding_group.py

```diff
@@ -26,30 +26,36 @@ def __init__(self, xml_tree):
         self.funding_fn = self.funding_group_object.fn_financial_information
 
     def funding_sources_exist_validation(self):
-        for funding in self.funding_sources or [None]:
-            funding_list = funding.get("funding-source") if funding else []
-            award_list = funding.get("award-id") if funding else []
-            funding_length = len(funding_list)
-            award_length = len(award_list)
-            is_valid = funding_length > 0 and award_length > 0
-            obtained = '{} values to <funding-source> and {} values to <award-id>'.format(funding_length, award_length)
-            advice = []
-            if funding_length == 0:
-                advice.append('<funding-source>')
-            if award_length == 0:
-                advice.append('<award-id>')
+        title = 'Funding source element validation'
+        validation_type = 'exist'
+        for funding in self.funding_group + self.funding_fn:
+            is_valid = False
+            advice = None
+            funding_list = funding.get("funding-source")
+            award_list = funding.get("award-id")
+            has_funding = len(funding_list) > 0
+            has_award = len(award_list) > 0
+            obtained = '{} values to funding source and {} values to award id'.format(len(funding_list), len(award_list))
```

Comment:
Trocar to por `for` ou `of` em  `values to funding` e `values to award`

Reply:
OK, @robertatakenaka